src/use-dropdown-menu.ts

```diff
@@ -78,6 +78,11 @@ export default function useDropdownMenu<ButtonElement extends HTMLElement = HTML
 			return;
 		}
 
+		// Initialize object to track if the removal happens before the addition of the event listener
```

Comment:
It's probably not immediately obvious why we're using an object instead of a boolean variable (especially for newer engineers), so let's add a succinct note underneath indicating why:

```suggestion
		// Initialize object to track if the removal happens before the addition of the event listener
		//  -> We're using an object here so that arrow functions below capture the reference and not the value
```